sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager_base.py

```diff
@@ -58,10 +141,14 @@ def _feature_flag_telemetry(
             feature_flag_reference = f"{endpoint}kv/{feature_flag.key}"
             if feature_flag.label and not feature_flag.label.isspace():
                 feature_flag_reference += f"?label={feature_flag.label}"
-            feature_flag_value[TELEMETRY_KEY][METADATA_KEY][FEATURE_FLAG_REFERENCE_KEY] = feature_flag_reference
-            feature_flag_value[TELEMETRY_KEY][METADATA_KEY][FEATURE_FLAG_ID_KEY] = self._calculate_feature_id(
-                feature_flag.key, feature_flag.label
-            )
+            if feature_flag_value[TELEMETRY_KEY].get("enabled"):
+                feature_flag_value[TELEMETRY_KEY][METADATA_KEY][FEATURE_FLAG_REFERENCE_KEY] = feature_flag_reference
+                feature_flag_value[TELEMETRY_KEY][METADATA_KEY][FEATURE_FLAG_ID_KEY] = self._calculate_feature_id(
+                    feature_flag.key, feature_flag.label
+                )
+                feature_flag_value[TELEMETRY_KEY][METADATA_KEY][ALLOCATION_ID_KEY] = self._generate_allocation_id(
```

Comment:
In .net we omit the field rather than setting it to false. Not sure if it really matters.

Reply:
I'm not sure what you mean with `false` this is the allocation id? It should always have a string value or `None`.

Reply:
Misspoke. Meant to say `None`. My concern is that setting this to none will cause it to be persisted like:

{
    "FeatureFlagId: "someId",
    "AllocationId": null
}

Rather than omitted

{
    "FeatureFlagId: "someId"
}